### PR TITLE
Add proposed ADR for Lit framework

### PR DESCRIPTION
**Pull request by heymatthenry**

### CODE REVIEW

File: decisions/0002-use-lit.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+There are several frameworks on offer that leverage the browser custom element APIs, any of which would be a strong choice as a basis for the next version of the design system, but there are two primary choices the team has been evaluating for this purpose:
+
+1. Stencil - Stencil is an all-in-one web framework that supports React-like component authoring. Stencil components are written using JSX and transpiled to native JavaScript web components. Further, Stencil provides custom tooling such as a [test runner](https://stenciljs.com/docs/testing/stencil-testrunner/overview), as well as tools for static site generation and server-side rendering functionality among others. Because it has its own integrated build system, it also makes it easy to incorporate other tools, like Sass/SCSS, with little or no additional configuration. Additionally, some agencies are already using Stencil to build web component implementations of and extensions to the Design System.
```
Comment (mejiaj) [inline decisions/0002-use-lit.md:13]: StencilJS can also target specific frameworks for compatibility.

See [Angular output target](https://stenciljs.com/docs/angular#adding-the-angular-output-target) example.

File: decisions/0002-use-lit.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+2. Lit - As with Stencil, there are existing USWDS-based federal website projects that leverage Lit.
```
Comment (mejiaj) [inline decisions/0002-use-lit.md:14]: Lit announced additional wrappers last year in this YouTube video - [Lit Labs: Framework wrapper generators](https://www.youtube.com/watch?v=CL_3TLFkwNE). You can see some listed here too under lit/labs https://github.com/lit/lit. 

It looks like this work is still experimental though.

**Review by mejiaj — COMMENTED** [review]
Reasoning is solid. I've added additional notes for StencilJS/Lit on framework integration.

I do believe the styling approach on Lit can be less flexible than StencilJS, but overall they are similar in that they're a light layer on top of the native APIs.

**Comment by heymatthenry** [comment]
> I do believe the styling approach on Lit can be less flexible than StencilJS, but overall they are similar in that they're a light layer on top of the native APIs.

I think that's right about the light layer on top of native APIs. I _think_ under the hood what's happening here:

https://github.com/uswds/uswds-next/blob/ff9102731831c554b95e5c06e917596e57523c3b/web-components/src/components/usa-banner/index.js#L23-L34

…is a mix of the adopted stylesheet (for the imported sass) and constructed stylesheet (for the CSS-in-looking stuff) APIs

Whichever way we end up going, we'll need to have a story about how styles are packaged and how we're going to use custom properties, but that's a separate issue.

File: decisions/0002-use-lit.md (hunk 2 above)
Comment (heymatthenry) [reply decisions/0002-use-lit.md:14]: Hopefully over time this is less of a thing. Even React is going to support Custom Elements [soon](https://github.com/facebook/react/issues/11347#issuecomment-1122275286)!

Whatever the answer ends up being to framework compat, it'll definitely need a lot of testing.

**Review by mejiaj — APPROVED** [review]
The performance and being so similar to the native API is really nice.

Also enjoying writing vanilla JS and the ability to choose between that and typescript is another win.

**Comment by mahoneycm** [comment]
I've been enjoying the Lit experience so far! I enjoy being able to write in vanilla JS over TS. The ability to pull in SCSS while we make style decisions has also made getting started pretty effortless. Using reactive properties has also made experimenting with variant attributes / classes a breeze 👍

**Comment by annepetersen** [comment]
Did we want to address whether we want to capitalize Web Components in here? (right now it's not, but other places we're doing so as standard to differentiate from USWDS components)

**Review by annepetersen — APPROVED** [review]
LGTM

**Pull request merged**